src/tensorlake/function_executor/allocation_runner/allocation_runner.py

```diff
@@ -245,6 +254,38 @@ def deliver_allocation_update(self, update: AllocationUpdate) -> None:
                 update=str(update),
             )
 
+    def run_request_context_operation(
+        self,
+        operation: (
+            PrepareWriteRequest
+            | PrepareReadRequest
+            | CommitWriteRequest
+            | FunctionProgressUpdateRequest
+        ),
+    ) -> (
+        PrepareWriteResponse
+        | PrepareReadResponse
+        | CommitWriteResponse
+        | FunctionProgressUpdateResponse
+    ):
+        """Runs the given request context operation and returns its result.
+
+        Blocks until the operation completes.
+        Raises exception on error.
+        """
+        if isinstance(operation, PrepareReadRequest):
+            return self._request_state.prepare_read(operation)
+        elif isinstance(operation, PrepareWriteRequest):
+            return self._request_state.prepare_write(operation)
+        elif isinstance(operation, CommitWriteRequest):
+            return self._request_state.commit_write(operation)
+        elif isinstance(operation, FunctionProgressUpdateRequest):
+            return self._allocation_progress.update(operation)
+        else:
+            raise InternalError(
```

Comment:
should this be a `RuntimeError` instead of `InternalError`?

Reply:
Sounds good. In this particular case we don't need to use SDK exceptions because these are not raised to user code here. So using RuntimeError is fine and a bit less misleading, cause ppl might assume we actually raise to user code here if they see SDK Exception (InternalError).